lib/datadog/open_feature/native_evaluator.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'json'
+require_relative 'ext'
+require_relative 'resolution_details'
+
+module Datadog
+  module OpenFeature
+    # Evaluation using native extension
+    class NativeEvaluator
+      def initialize(configuration)
+        @libdatadog_api_failure = Datadog::Core::LIBDATADOG_API_FAILURE
+        if configuration && !@libdatadog_api_failure
+          @configuration = Datadog::Core::FeatureFlags::Configuration.new(configuration)
+        end
+      end
+
+      def get_assignment(flag_key, default_value, context, expected_type)
+        return build_fatal_error(default_value) if @libdatadog_api_failure
+
+        configuration = @configuration
+        return build_not_ready_error(default_value) if configuration.nil?
+
+        native_details = configuration.get_assignment(flag_key, expected_type.to_sym, context)
```

Comment:
It was converted from symbols originally to the strings because you've said that it should be a string and now we converting it to the symbol, why? I can pass a symbol from the `EvaluationEngine` level instead and avoid those allocations, but why has it changed?

> Please keep expected flag types aligned with OF: "string" | "integer" | "float" | "boolean" | "object"

[Slack](https://dd.slack.com/archives/C0959U2JP2N/p1762965416657949?thread_ts=1762948575.383089&cid=C0959U2JP2N)